src/ucp/tag/rndv.c

```diff
@@ -16,12 +16,30 @@
 #include <ucp/proto/proto_am.inl>
 #include <ucs/datastruct/queue.h>
 
-static int ucp_rndv_is_get_zcopy(ucp_request_t *sreq, ucp_rndv_mode_t rndv_mode)
+static int ucp_rndv_is_get_zcopy(ucs_memory_type_t mem_type,
+                                 ucp_rndv_mode_t rndv_mode)
 {
     return ((rndv_mode == UCP_RNDV_MODE_GET_ZCOPY) ||
             ((rndv_mode == UCP_RNDV_MODE_AUTO) &&
-             (UCP_MEM_IS_HOST(sreq->send.mem_type) ||
-              UCP_MEM_IS_ROCM(sreq->send.mem_type))));
+              (UCP_MEM_IS_HOST(mem_type) || UCP_MEM_IS_ROCM(mem_type))));
+}
+
+static int ucp_rndv_is_recv_pipeline_needed(ucp_request_t *rndv_req,
+                                            ucs_memory_type_t mem_type)
+{
+    ucp_md_index_t md_index;
+    uct_md_attr_t *md_attr;
+
+    /* check if there is a bw lane to register mem type */
+    ucs_for_each_bit(md_index,
+                     ucp_ep_config(rndv_req->send.ep)->key.rma_bw_md_map) {
+        md_attr = &rndv_req->send.ep->worker->context->tl_mds[md_index].attr;
+        if (md_attr->cap.reg_mem_types & UCS_BIT(mem_type)) {
```

Comment:
q: does it mean that we will not use RNDV pipeline protocol for BW lanes with MEMTYPE registration capability (e.g. IB/GPUDirect)?

Reply:
Yes.

Reply:
> Yes.

👍